### ISSUE TITLE
Can't locate object method "render"

### ISSUE THREAD

**Issue by matt-couchman** [genomehubs/template-plugin#1]
Hi, we've taken a branch off template-plugin to use in our GenomeHub and we see this error when we load the home page (http://localhost/):

Unknown Exception:
  Can't locate object method "render" via package "EnsEMBL::Web::Document::HTML::AnnotatedSpecies" at /ensembl/ensembl-webcode/modules/EnsEMBL/Web/Controller/SSI.pm line 152.
  Thrown by EnsEMBL::Web::Controller::process in module EnsEMBL::Web::Apache::SSI at /ensembl/ensembl-webcode/modules/EnsEMBL/Web/Apache/SSI.pm on line 109
  Called by EnsEMBL::Web::Apache::SSI::handler in module EnsEMBL::Web::Apache::Handlers at /ensembl/ensembl-webcode/modules/EnsEMBL/Web/Apache/Handlers.pm on line 505
  Called by EnsEMBL::Web::Apache::Handlers::handler in module main at -e on line 0
  Called by (eval) in module main at -e on line 0 at /ensembl/ensembl-webcode/modules/EnsEMBL/Web/Exception.pm line 70.
ERROR: 2e74e82b86 (/index.html) (Server Exception)
Uncaught exception 'Unknown' with message 'Can't locate object method "render" via package "EnsEMBL::Web::Document::HTML::AnnotatedSpecies" at /ensembl/ensembl-webcode/modules/EnsEMBL/Web/Controller/SSI.pm line 152.
'
  Thrown by EnsEMBL::Web::Controller::process in module EnsEMBL::Web::Apache::SSI at /ensembl/ensembl-webcode/modules/EnsEMBL/Web/Apache/SSI.pm on line 109
  Called by EnsEMBL::Web::Apache::SSI::handler in module EnsEMBL::Web::Apache::Handlers at /ensembl/ensembl-webcode/modules/EnsEMBL/Web/Apache/Handlers.pm on line 505
  Called by EnsEMBL::Web::Apache::Handlers::handler in module main at -e on line 0
  Called by (eval) in module main at -e on line 0 at /ensembl/ensembl-webcode/modules/EnsEMBL/Web/Apache/ServerError.pm line 72.
REQUEST(G): [served at Thu Mar 28 16:40:07 2019 by 447 in 9ms] /index.html

I'm not sure if this means we have a missing file or we need to define our own render function or something else. Can you give any advice? Thanks.

**Comment by rjchallis** [comment]
Sorry for the delay, been a busy couple of weeks. It looks like I've been neglecting this repository when updating the rest of the code so it has become a little outdated.

I'll take a look at this and try to get an updated version working over the next week.

**Comment by matt-couchman** [comment]
Thanks for getting back to us. Would it be worthwhile for us to use older versions of the other software in order to get something working in the short term? Is there anything we could do to help to fix the issue?

**Comment by rjchallis** [comment]
It's taken a little longer than planned to get other projects out of the way to focus on this so, yes, running the Ensembl 89 based code should let you get a bit further as a short term fix. I'll finally be focussed back on GenomeHubs for a while from Monday so will try to get this working as soon as I can. 

It would be useful to see the `[REPOSITORIES]` and `[DATA_SOURCE]` sections of your `setup.ini` files to check that all the versions are compatible. Otherwise there were some changes to the Ensembl configuration between v89 and v93 so I likely just need to match the right bits back up to be able to provide a working example.

**Comment by matt-couchman** [comment]
Hi Richard,

Thanks very much we’ll have a try with Ensembl v89.

Thanks too for looking at our [REPOSITORIES] and [DATA_SOURCE] sections:

[REPOSITORIES]
  ENSEMBL_URL = https://github.com/Ensembl
  ENSEMBL_BRANCH = release/93

  BIOPERL_URL = https://github.com/bioperl
  BIOPERL_BRANCH = master

  API_PLUGIN_URL = https://github.com/EnsemblGenomes/ensemblgenomes-api
  API_PLUGIN_BRANCH = release/eg/40
  API_PLUGIN_PACKAGE = EG::API

  YOUR_PLUGIN_URL = https://git.nbi.ac.uk/rg-matthew-hartley/genomehubs-template-plugin.git
  YOUR_PLUGIN_BRANCH = master
  YOUR_PLUGIN_PACKAGE = EG::OpenPlantInsects

  GENOMEHUBS_PLUGIN_URL = https://github.com/genomehubs/gh-ensembl-plugin
  GENOMEHUBS_PLUGIN_BRANCH = 17.03
  GENOMEHUBS_PLUGIN_PACKAGE = EG::GenomeHubs

  EG_COMMON_PLUGIN_URL = https://github.com/EnsemblGenomes/eg-web-common
  EG_COMMON_PLUGIN_BRANCH = release/eg/40
  EG_COMMON_PLUGIN_PACKAGE = EG::Common

  PUBLIC_PLUGINS = [ ]

[DATA_SOURCE]
  SPECIES_DBS = [
    melitaea_cinxia_core_40_93_1
    myzus_persicae_mper1_core_40_93_1
  ]
  MULTI_DBS = [ ensembl_ontology_90 ensemblgenomes_info_40 ensembl_website_93 ]
  COMPARA_DBS = [ ensembl_compara_metazoa_40_93 ensembl_compara_pan_homology_40_93 ]

Does the YOUR_PLUGIN section create a Perl module from a template or should it be populated with something first?

Thanks,

                                Matt.

**Comment by rjchallis** [comment]
This could be a simple fix - you may be able to get your site up and running if you change the `GENOMEHUBS_PLUGIN_BRANCH` to `18.10` in your `[REPOSITORIES]` section, `17.03` is an older version that worked with Ensembl 85.

I'm just wotkrking through the gitbooks instructions to try to make sure everything works as intended this time around. If changing the GenomeHubs plugin branch doesn't fix this then I'll take a proper look once I get to the end of the rest of the instructions.

There is no need to create a module - the name you give `YOUR_PLUGIN_PACKAGE` is used as a hash key in a `Plugins,pm` module so the files in the plugin can be used in place of those in plugins that are further down the list.

**Comment by rjchallis** [comment]
Hi Matt

This took a little longer than I'd anticipated but I've gone through all the docs for the latest version and created new 19.05 repositories/containers that should now work together much more smoothly. You'll have to work back through the earlier steps to grab the latest versions but hopefully once you get back to linking the template repository it will work as it is supposed to.

Please do let me know if you spot any other bugs while working through the docs or have any suggestions of thing that can be improved.

Rich

**Comment by matt-couchman** [comment]
Hi Richard,

Thank very much for doing that, I’ll start updating our instance this afternoon.

There were a couple of things that might be helpful to update in the docs:


  1.  In the mysql page:

https://genomehubs.gitbook.io/genomehubs/quick-start/mysql-setup
[https://app.gitbook.com/share/space/thumbnail/-LeLgajqm8Cc4NCOpzXC.png]<https://genomehubs.gitbook.io/genomehubs/quick-start/mysql-setup>

2. Setup MySQL database server<https://genomehubs.gitbook.io/genomehubs/quick-start/mysql-setup>
/
genomehubs.gitbook.io



I use an additional argument to the docker command to add the "--max-packet-size" configuration without the need to run "docker exec".

docker run -d \
           --name genomehubs-mysql \
           --network genomehubs-network \
           -v /opt/genomehubs/mysql/data:/var/lib/mysql \
           -e MYSQL_ROOT_PASSWORD=CHANGEME \
           -e MYSQL_ROOT_HOST='172.16.0.0/255.240.0.0' \
           -p 3306:3306 \
           mysql/mysql-server:5.5 --max-allowed-packet=10000000000


2. I ran into problems with our chromosome names having underscores in them, e.g. "scaffold_1". If I remember correctly this was due to the underscore being used to separate out different pieces of metadata in some regular expression where it was expecting a full stop.

It might be worth adding a warning to this effect.

3. It would be great to have a bit more of an explanation of which of the assembly metadata fields are required and what they are used for. In some cases I've ran into trouble because something has not been set or it has not been set correctly. In particular, for debugging purposes, it would be useful to know how these variables correspond to columns in databases and URLs in the website.

4. It would also be useful to know which steps are essential. I have the impression that steps 5 (Run Analysis) and 6 (Import Analysis Results) are not required to setup a server for hosting genomes.

5. On a similar note it would be great to have some instructions on how to add data from analysis, e.g. BUSCO, that have been generated outside of the genomehubs docker environment

Thanks again for all your help.

Kind regards,

Matt.
________________________________
